include/commissioner/commissioner.hpp

```diff
@@ -194,90 +294,16 @@ class Commissioner
      */
     using PetitionHandler = std::function<void(const std::string *aExistingCommissionerId, Error aError)>;
```

Comment:
Why is `PetitionHandler` not also handled by `CommissionerHandler` ?

Reply:
Because petition is initiated by the user with the petition API:
```c++
virtual void Petition(PetitionHandler aHandler, const std::string &aAddr, uint16_t aPort) = 0;
```

Is it better to specify the callback handler when the user invoking the trigger?

Reply:
Seems reasonable to me.